thunderstorm_auth/decoder.py

```
@@ -26,7 +26,7 @@ def decode_token(token, jwks, leeway=DEFAULT_LEEWAY):
     try:
         key_id = get_signing_key_id_from_jwt(token)
 
-        public_key = get_public_key_from_jwk(jwks['keys'][key_id])
+        public_key = get_public_key_from_jwk(jwks.get('keys', {})[key_id])
```

Surely the JWK set itself being empty should be a hard failure, shouldn't it? If it needs to be validated I would have said that should happen where it's loaded to prevent the app from even starting with a bad JWKS.

---

There is an assertion in the helper which loads the file.
https://github.com/artsalliancemedia/thunderstorm-auth-library/blob/master/thunderstorm_auth/utils.py#L14

---

👍 so can we assume that it's there by this point and fail hard if it's not?

---

Is that helper just for tests though? i.e. when users are supplying JWKs to the auth library we don't know if they will pull those from a file or if they will have just received them directly from the user service JWKs endpoint or just created a dict of random junk and passed that in as their JWK set. I added another commit now which checks to make sure 'keys' is not empty just before trying to decode a token. - https://github.com/artsalliancemedia/thunderstorm-auth-library/pull/17/commits/f0bc90f80f17bd1f1c3b65dd2714c518bcebb620